Include/internal/pycore_hamt.h

```diff
@@ -5,7 +5,7 @@
 #  error "this header requires Py_BUILD_CORE define"
 #endif
 
-#define _Py_HAMT_MAX_TREE_DEPTH 7
+#define _Py_HAMT_MAX_TREE_DEPTH 8
```

Comment:
It lacks an important comment like: "8 should be enough for everyone", famous last words.

Reply:
good point actually, i added an explainer